backends/gstreamer/webrtc.rs

```diff
@@ -123,14 +123,16 @@ impl WebRtcControllerBackend for GStreamerWebRtcController {
     }
 
     fn add_stream(&mut self, stream_id: &MediaStreamId) -> WebrtcResult {
-        let stream =
-            get_stream(stream_id).expect("Media streams registry does not contain such ID");
-        let mut stream = stream.lock().unwrap();
-        let stream = stream
-            .as_mut_any()
-            .downcast_mut::<GStreamerMediaStream>()
-            .ok_or("Does not currently support non-gstreamer streams")?;
-        stream.insert_capsfilter();
+        {
+            let stream =
```

Comment:
This does mean we end up locking twice. I could alternatively pass down the `&mut GstreamerMediaStream` to `link_stream` directly, but that gets uglier